checker/tests/optional/EnsuresPresentIfTest.java

```diff
@@ -0,0 +1,26 @@
+import java.util.Optional;
+import org.checkerframework.checker.optional.qual.*;
+import org.checkerframework.checker.optional.qual.EnsuresPresentIf;
+
+public class EnsuresPresentIfTest {
+
+  // :: warning: (optional.field)
+  private Optional<String> optId = Optional.of("abc");
+
+  public @org.checkerframework.dataflow.qual.Pure Optional<String> getOptId() {
+    return Optional.of("abc");
+  }
+
+  @EnsuresPresentIf(result = true, expression = "getOptId()")
+  public boolean hasPresentId1() {
+    return getOptId().isPresent();
+  }
+
+  void client() {
+    if (hasPresentId1()) {
+      optId.get();
```

Comment:
@mernst this line reports `error: (method.invocation)` but I am not sure why. I think I am annotating `hasPresentId1` correctly.

Reply:
This line should have been `getOptId().get()`.  Then the annotation works.  I added more test code to illustrate.

Reply:
Right, that makes sense. This test would have passed if I instead wrote
```
expression="optId"
```
Right?

Reply:
Correct.  I added such a test.